### PR TITLE
feat(flat-components): add scenes controller components

### PR DESCRIPTION
**Pull request by Cheerego7**

### CODE REVIEW

File: packages/flat-components/src/components/ClassroomPage/ScenesController/index.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import preScenesSVG from "./image/previous-scene.svg";
+
+import React, { FC } from "react";
+import classNames from "classnames";
+
+export interface ScenesControllerProps {
+    addScene: () => void;
+    preScene: () => void;
+    nextScene: () => void;
+    currentScene: number;
```
Comment (BlackHole1) [inline packages/flat-components/src/components/ClassroomPage/ScenesController/index.tsx]: currentSceneIndex

File: packages/flat-components/src/components/ClassroomPage/ScenesController/index.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+export const ScenesController: FC<ScenesControllerProps> = ({
+    addScene,
+    preScene,
+    nextScene,
+    currentScene,
+    scenesCount,
+    disabled,
+}) => {
+    const isFirstScene = currentScene === 0;
+    const isLastScene = currentScene + 1 === scenesCount;
```
Comment (BlackHole1) [inline packages/flat-components/src/components/ClassroomPage/ScenesController/index.tsx]: isEndScene

File: packages/flat-components/src/components/ClassroomPage/ScenesController/style.less [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    box-shadow: 0 4px 12px 0 rgb(0 0 0 / 8%);
+    background-color: white;
+}
+
+.scenes-controller-container-disabled {
+    opacity: 0.5;
+    * {
+        cursor: not-allowed;
+    }
+}
```
Comment (BlackHole1) [inline packages/flat-components/src/components/ClassroomPage/ScenesController/style.less]:
```suggestion
    &.disabled {
        opacity: 0.5;
        * {
            cursor: not-allowed;
        }
    }
}
```

File: packages/flat-components/src/components/ClassroomPage/ScenesController/index.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    return (
+        <div
+            className={classNames("scenes-controller-container", {
+                "scenes-controller-container-disabled": disabled,
```
Comment (BlackHole1) [inline packages/flat-components/src/components/ClassroomPage/ScenesController/index.tsx]:
```suggestion
                disabled,
```

File: packages/flat-components/src/components/ClassroomPage/ScenesController/index.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            })}
+        >
+            <div className="scenes-controller-btn-list">
+                <div className="scenes-controller-btn" onClick={disabled ? undefined : addScene}>
```
Comment (BlackHole1) [inline packages/flat-components/src/components/ClassroomPage/ScenesController/index.tsx]:
```ts
const warpOnClick = useCallback(fn => {
    if (disabled) {
        return () => {};
    }
    
    return fn();
}, disabled);

onClick={warpOnClick(addScene)}
```

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

File: packages/flat-components/src/components/ClassroomPage/ScenesController/index.tsx (hunk 2 above)
Comment (Cheerego7) [reply packages/flat-components/src/components/ClassroomPage/ScenesController/index.tsx]: there is no ambiguity

Comment (BlackHole1) [reply packages/flat-components/src/components/ClassroomPage/ScenesController/index.tsx]: `First` and `End` match better

**Review by BlackHole1 — APPROVED** [review]

**Pull request merged**